clang/include/clang/AST/DeclTemplate.h

```diff
@@ -1634,6 +1640,14 @@ class TemplateTemplateParmDecl final
   using TemplateParmPosition::setPosition;
   using TemplateParmPosition::getIndex;
 
+  /// Whether this template template parameter was declared with
+  /// the 'typename' keyword.
+  bool wasDeclaredWithTypename() const { return Typename; }
```

Comment:
So these should return the syntax of typename and class rather than a bool.

Reply:
We do not have an enum for that yet afaik (ie this is consistent with ActOnTypeParameter).
I would not oppose using an enum everywhere